released/discovery_center/mission_4356/step1/main.tf

```diff
@@ -1,109 +1,111 @@
 }
 
-###############################################################################################
+# ------------------------------------------------------------------------------------------------------
 # Assignment of users as sub account administrators
-###############################################################################################
+# ------------------------------------------------------------------------------------------------------
 resource "btp_subaccount_role_collection_assignment" "subaccount-admins" {
-  for_each             = toset("${var.subaccount_admins}")
-  subaccount_id        = data.btp_subaccount.project.id
+  for_each             = toset(var.subaccount_admins)
+  subaccount_id        = data.btp_subaccount.dc_mission.id
   role_collection_name = "Subaccount Administrator"
   user_name            = each.value
 }
 
-###############################################################################################
+# ------------------------------------------------------------------------------------------------------
 # Assignment of users as sub account service administrators
-###############################################################################################
+# ------------------------------------------------------------------------------------------------------
 resource "btp_subaccount_role_collection_assignment" "subaccount-service-admins" {
-  for_each             = toset("${var.subaccount_service_admins}")
-  subaccount_id        = data.btp_subaccount.project.id
+  for_each             = toset(var.subaccount_service_admins)
+  subaccount_id        = data.btp_subaccount.dc_mission.id
   role_collection_name = "Subaccount Service Administrator"
   user_name            = each.value
 }
 
-######################################################################
-# Extract list of CF landscape labels from environments
-######################################################################
+# ------------------------------------------------------------------------------------------------------
+# CLOUDFOUNDRY PREPARATION
+# ------------------------------------------------------------------------------------------------------
+#
+# Fetch all available environments for the subaccount
 data "btp_subaccount_environments" "all" {
-  subaccount_id = data.btp_subaccount.project.id
+  subaccount_id = data.btp_subaccount.dc_mission.id
 }
-
-locals {
-  cf_landscape_labels = [
-    for env in data.btp_subaccount_environments.all.values : env.landscape_label
-    if env.environment_type == "cloudfoundry"
-  ]
+# ------------------------------------------------------------------------------------------------------
+# Take the landscape label from the first CF environment if no environment label is provided
+# (this replaces the previous null_resource)
+# ------------------------------------------------------------------------------------------------------
+resource "terraform_data" "replacement" {
```

Comment:
maybe a better resource name :) ? .. `landscape_label`...`obtained_landscape_label`...`effective_label`..picked_up_label`...or something

Reply:
That makes total sense. Thanks!